crates/forge/src/warn.rs

```diff
@@ -57,28 +56,15 @@ pub(crate) async fn warn_if_incompatible_rpc_version(
             Url::parse(url).with_context(|| format!("could not parse url: {url}"))?,
         ));
 
-        handles.push(async move {
-            (
-                client
-                    .spec_version()
-                    .await
-                    .map(|version| {
-                        version
-                            .parse::<Version>()
-                            .with_context(|| format!("could not parse version: {version}"))
-                    })
-                    .context("error while calling rpc node"),
-                url,
-            )
-        });
+        handles.push(async move { (get_and_parse_spec_version(&client).await, url) });
     }
 
     // assert version
     for handle in handles {
         let (version, url) = handle.await;
-        let version = version??;
+        let version = version?;
 
-        if !expected_version.matches(&version) {
+        if !is_supported_version(&version) {
```

Comment:
I wonder what happens when `--json` flag is passed? Since the warnings aren't intended to abort the command, I think storing those in some `warnings` sequence would be better.

Reply:
When the `--json` flag is passed, it is ignored by this warning. If we decide to modify this behavior, we also need to adjust this for the other [warning](https://github.com/foundry-rs/starknet-foundry/blob/master/crates/sncast/src/starknet_commands/script.rs#L261) about the `sncast_std` version that appears when executing the script command.

Reply:
I agree, ideally warnings should be printed to stderr, to allow users to filter them. Let's do this (all of them use the same function, so this should be rather easy).

I believe, however, that the issue is may not be isolated to these particular cases. Now this should be verified with scarb team, but it seems some of the commands (like declare) may also break json, because scarb returns multiple objects (only last json is from sncast):

```
{"status":"compiling","message":"lib(map) map v0.1.0 (/Users/wojciech/git-repos/starkware/starknet-foundry/crates/sncast/tests/data/contracts/map/Scarb.toml)"}
{"type":"warn","message":"Sierra, textual Sierra and CASM lib targets have been disabled, Scarb will not produce anything"}
{"status":"compiling","message":"starknet-contract(map) map v0.1.0 (/Users/wojciech/git-repos/starkware/starknet-foundry/crates/sncast/tests/data/contracts/map/Scarb.toml)"}
{"status":"finished","message":"release target(s) in 1 second"}
{"class_hash":"0x43a9554442216b8c5b551c6cae6357ce69fd9838e9bf40fb27b455cac5ba9f4","transaction_hash":"0x21d5dfe1854c294b371632ea9f2435017d0373f8adc6d4194c88a16c87eb0c","command":"declare"}
```

we should take a closer look at this under different issue. We must also decide what to do with our `--wait` function (under the same issue), as it also uses `println` to inform about tx hash, so it can also potentially break json. Let me know if you will make one, or if you want me to do this.

Reply:
Discussed offline with @THenry14 that he will create an issue for this and link it here.

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/1723